xo-archiver

```diff
@@ -442,14 +487,89 @@ function copy_to_s3 {
     if aws s3 cp $WORKING_DIR/ $_remote_xva_path/ --recursive
     then
         log "Successfully copied exports of \"$_vm_name\" on bucket \"$AWS_BUCKET_XO_ARCHIVES\""
-        log "Cleaning up $WORKING_DIR/"
-        rm -Rf $WORKING_DIR/
+        clean_working_dir
     else
         die "Fail to push to s3"
     fi
 }
 
 
+# Usage: copy_from_s3 <VM-NAME>
+# copies the <VM-NAME> corresponding files from the s3 bucket specified through envvar
+function copy_from_s3 {
+    if [[ -z $1 ]]
+    then
+        die "copy_from_s3 needs parameter: VM-NAME. E.g: foo-bar-01"
+    fi
+
+    # ensure bucket var is set
+    if [[ -z "$AWS_BUCKET_XO_ARCHIVES" ]]
+    then 
+        die "Bucket is undefined. look at the --help and search for AWS_BUCKET_XO_ARCHIVES"
+    fi
+
+    create_working_dir
+
+    local _remote_xva_path="s3://$AWS_BUCKET_XO_ARCHIVES/$1"
+    log "Copying archives from \"$_remote_xva_path\" into $WORKING_DIR/"
+
+    if aws s3 cp $_remote_xva_path/ $WORKING_DIR/ --recursive --force-glacier-transfer
+    then
+        if ls -A $WORKING_DIR/ | grep -q "[a-Z0-9]"
+        then
+            log "Successfully copied files from bucket \"$_remote_xva_path\" into $WORKING_DIR/"
+        else
+            clean_working_dir
+            die "Nothing to copy from S3, ensure \"$_remote_xva_path\" exists and contains files"
+        fi
+    else
+        clean_working_dir
+        die "Fail to copy from S3"
+    fi
+}
+
+
+# Usage: get_metadata_from_local_files
+# Checks that local files contain mandatory informations
+# and exports vars for futur import: SR-ID and a readable XVA-file
+# This function doesn't need argument as $WORKING_DIR is used
+# and this var depends of the selector used to load this script
+function get_metadata_from_local_files {
+    if ! ls -A $WORKING_DIR/ | grep -q "[a-Z0-9]"
+    then
+        die "No file found in local directory \"$WORKING_DIR/\""
+    fi
+
+    # this exported envvar is used in Main to call import_vm when restoring
+    export VM_SR_ID="$(cat $WORKING_DIR/SR-ID.txt 2>/dev/null)"
+
+    if echo $VM_SR_ID | grep -q "[a-Z0-9]"
+    then
+        log "Found Storage Repository ID \"$VM_SR_ID\" from file $WORKING_DIR/SR-ID.txt"
+    else
+        unset $VM_SR_ID
+        die "Unable to export Storage Repository ID from file $WORKING_DIR/SR-ID.txt"
+    fi
+
+    # Try to get xva file from VM.json
+    local _archived_vm_id=$(cat $WORKING_DIR/VM.json 2>/dev/null | jq -r --exit-status '.[0].id')
+    if echo $_archived_vm_id | grep -q "[a-Z0-9]"
```

Comment:
You should add a GHA pipeline for shell checks :wink: You'll find plenty of stuff to improve I suppose, especially with shell variables not between double quotes :wink: